src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -578,6 +590,199 @@ public static void registerTESRItemStack(Item item, int metadata, Class<? extend
+            // Values must be multiplied by 16, divided by 4 for average => x4
+            blockLight *= 4;
+            skyLight *= 4;
+        }
+        
+        // Dummy overrides
+
+        @Override
+        public void setQuadTint(int tint) {}
+
+        @Override
+        public void setQuadOrientation(EnumFacing orientation) {}
+
+        @Override
+        public void setApplyDiffuseLighting(boolean diffuse) {}
+
+        @Override
+        public void setTexture(TextureAtlasSprite texture) {}
+    }
+    
+    private static final LightGatheringTransformer lightGatherer = new LightGatheringTransformer();
+
+    public static void renderLitItem(RenderItem ri, IBakedModel model, int color, ItemStack stack) 
+    {
+        BufferBuilder bufferbuilder = Tessellator.getInstance().getBuffer();
+
+        List<BakedQuad> allquads = new ArrayList<>();
+
+        for (EnumFacing enumfacing : EnumFacing.VALUES)
+        {
+            allquads.addAll(model.getQuads(null, enumfacing, 0));
+        }
+
+        allquads.addAll(model.getQuads(null, null, 0));
+
+        // Current list of consecutive quads with the same lighting
+        List<BakedQuad> segment = new ArrayList<>();
+
+        // Lighting of the current segment
+        int segmentBlockLight = -1;
+        int segmentSkyLight = -1;
+        int segmentColorMultiplier = color;
+        
+        // Tint index cache to avoid unnecessary IItemColor lookups
+        int prevTintIndex = -1;
+
+        for (int i = 0; i < allquads.size(); i++) 
+        {
+            BakedQuad q = allquads.get(i);
+
+            // Lighting of the current quad
+            int bl = 0;
+            int sl = 0;
+
+            // Fail-fast on ITEM, as it cannot have light data
+            if (q.getFormat() != DefaultVertexFormats.ITEM) 
+            {
+                lightGatherer.setVertexFormat(q.getFormat());
+                if (lightGatherer.hasLighting())
+                {
+                    q.pipe(lightGatherer);
+                }
+                bl = lightGatherer.blockLight;
+                sl = lightGatherer.skyLight;
+            }
+            
+            int colorMultiplier = segmentColorMultiplier;
+            // If there is no color override, and this quad is tinted, we need to apply IItemColor
+            if (color == 0xFFFFFFFF && q.hasTintIndex())
+            {
+                int tintIndex = q.getTintIndex();
+                
+                if (prevTintIndex != tintIndex)
+                {
+                    colorMultiplier = getColorMultiplier(stack, tintIndex);
+                }
+                prevTintIndex = tintIndex;
+            }
+            else
+            {
+                colorMultiplier = color;
+                prevTintIndex = -1;
+            }
+
+            // If this is a new light value, draw the segment and flush it
+            if (segmentBlockLight != bl || segmentSkyLight != sl || segmentColorMultiplier != colorMultiplier) 
```

Comment:
There's an issue here - the problem with passing the colour multiplier directly to `renderQuads` in `drawSegment` in this way is that it requires all quads in the segment to have the same colour multiplier.

This means that the code here is splitting the quads into more segments than would otherwise be needed. As this results in more GL calls, I feel it works out as an overall performance regression, despite the `IItemColor` savings.

Of particular note, this also imposes an overhead on vanilla items now, as while there are no vanilla items with emissive lighting, there are some with multiple colour tints, which now need multiple draw calls.